### PR TITLE
esp8266 softap support

### PR DESCRIPTION
**Pull request by wuwx**
issue #46 softap support

### CODE REVIEW

File: src/inet/network_interface/wifi_softap_espressif.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
+#include "wifi_softap_espressif.h"
```
Comment (eerimoq) [inline src/inet/network_interface/wifi_softap_espressif.c]: Add a file header.

File: src/inet/network_interface/wifi_softap_espressif.h [hunk 2]
```diff
@@ -0,0 +1,23 @@
+#ifndef __INET_NETWORK_INTERFACE_WIFI_SOFTAP_ESPRESSIF_H__
```
Comment (eerimoq) [inline src/inet/network_interface/wifi_softap_espressif.h]: Add file header and function comments.

File: src/simba.mk [hunk 3]
```diff
@@ -183,6 +183,7 @@ INET_SRC_TMP ?= \
 
 ifeq ($(ARCH), esp)
 INET_SRC_TMP += network_interface/wifi_station_espressif.c
```
Comment (eerimoq) [inline src/simba.mk]: Replace with 

```
INET_SRC_TMP += network_interface/wifi_station_espressif.c \
                network_interface/wifi_softap_espressif.c
```

File: src/inet/network_interface/wifi_softap_espressif.c [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERSOCKTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * This file is part of the Simba project.
+ */
+
+#include "simba.h"
+
+#if defined(ARCH_ESP)
```
Comment (eerimoq) [inline src/inet/network_interface/wifi_softap_espressif.c]: This ifdef can be removed, right? This file is only compiled for the ESP.

File: src/inet/network_interface/wifi_softap_espressif.c [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+    struct network_interface_wifi_softap_espressif_t *self_p;
+
+    self_p = container_of(netif_p,
+                          struct network_interface_wifi_softap_espressif_t,
+                          network_interface);
+
+    return (network_interface_wifi_softap_espressif_start(self_p));
+}
+
+static int stop(struct network_interface_t *netif_p)
```
Comment (eerimoq) [inline src/inet/network_interface/wifi_softap_espressif.c]: Is there any espressif function to stop the interface and disable the DHCP server?

**Comment by wuwx** [comment]
if esp12e startup softap interface
test code need some other device join the softap
i don't know how to simulate wifi station client

**Comment by eerimoq** [comment]
Ahh yeah that's tricky. Let's write a testcase that just starts and stops the SoftAP interface. Maybe check that the interface has an IP address after starting the interface. That will at least make sure that the code does not crash =)

**Comment by wuwx** [comment]
there is no network_interface_stｏｐ defined
so the test code don't include *network_interface_stop_t function

**Comment by eerimoq** [comment]
That's fine. I'll test this in a few days and merge if it works as expected =)

**Comment by eerimoq** [comment]
I've changed the implementation of the ESP WiFi wrapper. Instead of implementing the wrapper as a network interface it is now implemented as a driver. I intend to remove the Espressif WiFi station network interface and instead just have a general WiFi network interface that in the future can be used for other WiFi modules as well. See this commit for the driver implementation and examples:

**Comment by wuwx** [comment]
waiting for new implementation
it's a nice job~~

**Comment by eerimoq** [comment]
Pushed! I hope it is better than the last implementation =)

**Comment by wuwx** [comment]
greate!
i love it!

**Pull request closed without merging**